core/src/main/java/org/apache/carbondata/scan/result/iterator/DetailQueryResultIterator.java

```diff
@@ -45,10 +47,28 @@
 
   public DetailQueryResultIterator(List<BlockExecutionInfo> infos, QueryModel queryModel) {
     super(infos, queryModel);
+    this.queryModel = queryModel;
+  }
+
+  private Boolean flag;
+
+  private Long total = 0L;
+
+  private QueryModel queryModel;
+
+  @Override public boolean hasNext() {
+    flag = super.hasNext();
```

Comment:
in hasnext: i need to judge when  will it traverse the iterator over
in next: i need to accumulate the total time
but AbstractDetailQueryResultIterator didn't define 'next'